doc/neomake.txt

```diff
@@ -88,12 +88,16 @@ All 'args' will be |expand()|ed: >
         \ 'args': ['%:p'],
         \ }
 
-The 'exe' and 'args' entries can be a function (callback) or a dictionary with
-an expected entry 'fn' (which has to be a function).
-The function has to return a string, and receives no arguments.
+The 'exe' and 'args' entries can be a function reference (see |:h Funcref|) or
+a dictionary with an expected entry 'fn' (which has to be a function).
+The function must return a string, and the function will be called with no
+arguments.
 
-Running "|:Neomake!| lint" from file.txt would cause "lint /path/to/file.txt" to
-be run.
+In the above example, running "|:Neomake!| lint" from file.txt would cause
+"lint /path/to/file.txt" to be run. Do note that this is atypical usage and is
+provided as an example of args expansion only, since "|:Neomake!|" commands
+typically don't require the filename. The preferred method of running Neomake
+against the current file is "|:Neomake|" without the exclamation mark.
```

Comment:
This example in general was a poor choice since in involves misusing `:Neomake!`

What do you think about removing it entirely?

The fact that args are expanded is already mentioned earlier https://github.com/davidosomething/neomake/blob/cf62376ea2c7ab6d7f975971b15371115aba57cc/doc/neomake.txt#L85-L89